test/topology_experimental_raft/test_tablets.py

```diff
@@ -29,6 +29,19 @@ async def inject_error_on(manager, error_name, servers):
     await asyncio.gather(*errs)
 
 
+async def rolling_restart(manager, servers):
+    for idx, s in enumerate(servers):
+        await manager.server_restart(s.server_id, wait_others=(len(servers)-1))
+        # Wait for other servers to see the restarted server.
+        # Otherwise, the next server we are going to restart may not yet see "s" as restarted
+        # and will not send graceful shutdown message to it. Server "s" may learn about the
+        # restart from gossip later and close connections while we already sent CQL requests
+        # to it, which will cause them to time out. Refs #14746.
+        for idx2 in range(len(servers)):
+            if idx2 != idx:
+                await manager.server_sees_other_server(servers[idx2].ip_addr, s.ip_addr)
```

Comment:
Hmm... but what guarantees that this won't immediately finish, just because `servers[idx2].ip_addr` didn't yet observe that `s` had shutdown?

Reply:
Unlike the UP status, shutdown status is propagated on shutdown via direct RPCs to the nodes currently considered as UP, followed by 2s sleep. It's not guaranteed that it will propagate, although in the absence of network issues it is unlikely that it won't.

A more reliable way would be to have a barrier also after the node is stopped.

I wonder if this can be done on the framework level inside `manager`, but I'm not familiar with the framework enough.

Reply:
Easy, `manager.server_not_sees_other_server`.
And you would have to use `manager.server_stop` + `manager.server_start` instead of `manager.server_Restart`.

Reply:
Yes, I meant if manager.server_restart() could do it, so that the framework is safe by default.